camayoc/tests/qpc/api/v1/utils.py

```diff
@@ -12,6 +13,108 @@
     FailedScanException,
     WaitTimeError,
 )
+from camayoc.qpc_models import (
+    Credential,
+    Scan,
+    Source
+)
+
+
+def get_source(source_type, cleanup):
+    """Retrieve a single network source if available from config file.
+
+    Retreive one network source from the config file.
+    Sources listed under the following section are assumed to be available
+    on demand and do not require their power state to be managed.
+
+    The expected configuration in the Camayoc's configuration file is as
+    follows::
+
+        qpc:
+        # other needed qpc config data
+            sources:
+                 - hosts:
+                       - '127.0.0.1'
+                   name: local
+                   type: network
+                   credentials:
+                       - root
+
+    The credential listed is assumed to be in the top level 'credentials'
+    section.
+
+    This source is meant to be used for tests where we do not care about the
+    results of the scan, for example tests that assert we can pause and restart
+    a scan.
+
+    :returns: camayoc.qpc_models.Source that has been created on server and has
+        all credentials listed in config file created and associtated with it.
+    """
+    cfg = config.get_config()
+    cred_list = cfg.get('credentials', [])
+    src_list = cfg.get('qpc', {}).get('sources', [])
+    config_src = {}
+    if not (src_list and cred_list):
+        return
+    for src in src_list:
+        if src.get('type') == source_type:
+            config_src = src
+    if config_src:
+        config_src.setdefault('credential_ids', [])
+        src_creds = config_src.get('credentials')
+        for cred in src_creds:
+            for config_cred in cred_list:
+                if cred == config_cred['name']:
+                    server_cred = Credential(
+                        cred_type=source_type,
+                        username=config_cred['username'],
+                    )
+                    if config_cred.get('password'):
+                        server_cred.password = config_cred['password']
+                    else:
+                        server_cred.ssh_keyfile = config_cred['sshkeyfile']
+
+                    server_cred.create()
+                    cleanup.append(server_cred)
+                    config_src['credential_ids'].append(server_cred._id)
+        server_src = Source(
+            hosts=config_src['hosts'],
+            credential_ids=config_src['credential_ids'],
+            source_type=source_type,
+        )
+
+        if config_src.get('options'):
+            server_src.options = config_src.get('options')
+
+        server_src.create()
+        cleanup.append(server_src)
+    return server_src
```

Comment:
This will fail when there is no source of the source_type on the config file. The if block above won't execute and thus the server_src object won't be created. You have two options return None if no config_src is found or indent this line